hw02-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,49 @@
+package homework;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+
+    //todo: 3. надо реализовать методы этого класса
+
+    private final TreeMap<Customer, String> customerStringTreeMap;
+
+    public CustomerService() {
+
+        this.customerStringTreeMap = new TreeMap<>();
+
+    }
+
+    public Map.Entry<Customer, String> getSmallest() {
+
+        return Map.entry(new Customer(
+                        customerStringTreeMap.firstEntry()
+                                .getKey()
+                                .getId(),
+                        customerStringTreeMap.firstEntry()
+                                .getKey()
+                                .getName(),
+                        customerStringTreeMap.firstEntry()
+                                .getKey()
+                                .getScores()),
+                customerStringTreeMap.firstEntry().getValue());
```

Comment:
Лучше общие части вынести в локальную переменную:

```suggestion
        Customer key = customerStringTreeMap.firstEntry().getKey();
        String value = customerStringTreeMap.firstEntry().getValue();
        return Map.entry(new Customer(key.getId(), key.getName(), key.getScores()), value);
```